crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,430 @@
+use crate::{utils::is_node_equal, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{self, js_binary_expression, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsStatement, JsBinaryExpression, JsBinaryOperator,
+    JsCallExpression, JsLanguage, JsLogicalExpression, JsLogicalOperator, JsSyntaxKind,
+    JsUnaryOperator, JsYieldArgument, JsYieldExpression, T,
+};
+use biome_rowan::{
+    AstNode, BatchMutationExt, NodeOrToken, SyntaxTriviaPiece, TriviaPieceKind, WalkEvent,
+};
+
+declare_rule! {
+    /// Disallow the use of yoda expressions.
+    ///
+    /// A Yoda expression is a programming style where the two parts of the expression are reversed from the typical order in a conditional statement.
+    /// A Yoda expression places the constant portion of the expression on the left side of the conditional statement. They can be confusing to some people,
+    /// the rule forbids the use of it to improve code readability.
+    ///
+    /// The rule have an exception for range expressions, expressions like `0 < value && value < 1` or `value <= 0 || 1 < value` are allowed.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if ("red" == value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (true === value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (5 != value) {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (value === "red") {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value === value) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value != 5) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (0 < value && value < 1) {}
+    /// ```
```

Comment:
Let's put the Wikipidia link at the bottom, like this: 

https://github.com/biomejs/biome/blob/f685ebc6945001f2dcb7387e6157a158eb6e9aef/crates/biome_js_analyze/src/lint/a11y/no_access_key.rs#L34-L37

Reply:
done!